File: csharp/project/Console/ConsoleApp1/Exercism/Clock.cs
using System;

public class Clock : IEquatable<Clock>
{
    public int hours { get; set; }
    public int minutes { get; set; }
    public Clock(int hours, int minutes)
    {
        var totalMinutes = hours * 60 + minutes;
        TimeSpan Time = TimeSpan.FromMinutes(totalMinutes >= 0 ? totalMinutes : 1440 - (-totalMinutes % 1440));
        this.hours = Time.Hours;
        this.minutes = Time.Minutes;
    }

    public Clock Add(int minutesToAdd)
    {
        TimeSpan currentTime = (new TimeSpan(hours, minutes, 0)).Add(new TimeSpan(0, minutesToAdd, 0));
        return new Clock(currentTime.Hours, currentTime.Minutes);
        
    }

    public Clock Subtract(int minutesToSubtract)
    {
        TimeSpan currentTime = (new TimeSpan(hours, minutes, 0)).Subtract(new TimeSpan(0, minutesToSubtract, 0));
        return new Clock(currentTime.Hours, currentTime.Minutes);
    }


    public override string ToString()
    {
        return new TimeSpan(hours, minutes, 0).ToString(@"hh\:mm");
    }

    public bool Equals(Clock other)
    {
        return (hours == other.hours && 
                minutes == other.minutes && 
                other != null);
        
    }
}

File: javascript/clock/clock.js
//
// This is only a SKELETON file for the 'Clock' exercise. It's been provided as a
// convenience to get you started writing code faster.
//




export class Clock {
  constructor(hours, minutes = 0) {
    var totalMinutes = (((minutes + hours * 60) % 1440) + 1440) % 1440;
    this.hours = Math.floor(totalMinutes / 60);
    this.minutes = totalMinutes % 60;
  }

  toString() {
    return `${this.hours.toString().padStart(2, 0)}:${this.minutes.toString().padStart(2, 0)}`;
  }

  plus(minutesToAdd = 0) {
    return new Clock(this.hours, this.minutes + minutesToAdd);
  }

  minus(minutesToSubtract = 0) {
    return new Clock(this.hours, this.minutes - minutesToSubtract);
  }

  equals(clock) {
    return (clock.hours == this.hours &&
            clock.minutes == this.minutes);
  }
}
